src/query/query.pest

```diff
@@ -10,17 +10,24 @@ expression         =  { expr }
 funcNames          = _{ latest }
 latest             =  { "latest" }
 
-infixExpression = { firstArg ~ infixFunction ~ secondArg }
+infixExpression = { firstArg ~ infixFunction ~ lookupValue }
 firstArg        = { lookup }
-secondArg       = { string }
-infixFunction   = { char* }
+infixFunction   = { char{1, 2} }
```

Comment:
I wonder if this will cause problems? maybe something for the future but if you write `id = 21` it will have the infix function be chars "=" and "2" since you allow alphanumeric chars in char and then the lookup value will be "1" and this would be valid, there is a check in `query_parse` for this so not a big deal but i wonder if better definition of infixFunction allows us to better provide suggestions to the user like if it only captures "=" and then "21" instead of ["=", "2"] and then "1" we can probably say did you mean "==", "21" otherwise we it would be messy to find out the number was "21". Again definitely for the future!

Reply:
Ah good spot, actually would be ok with `id = 21` as it is sensitive to whitespace but would be a problem with `id=23` parsing as infix `=2` value `3`. Had left this bit deliberately relaxed and was managing the error handling in `query_parse` as you noticed. In general there is a bit of trickiness with this, do we use a really restrictive grammar and use the inbuilt pest errors (which are ok but not amazing) or do we have a less restrictive grammar but then return some more user-friendly errors in the parsing code. I've not thought that through properly yet as it is quite hard, so there is a bit of a mix here with this infixFunction being an example of a very permissive bit of a grammar. 

I think you're right though! It's more permissive than it needs to be, I've tightened that up for now.